Firestore/Source/Local/FSTPersistence.h

```diff
@@ -99,6 +100,31 @@ NS_ASSUME_NONNULL_BEGIN
  */
 - (void)commitGroup:(FSTWriteGroup *)group;
 
+@property (nonatomic, readonly, assign) const FSTTransactionRunner& run;
+
 @end
 
+@protocol FSTTransactional
+
+- (void)startTransaction;
+
+- (void)commitTransaction;
+
+@end
+
+struct FSTTransactionRunner {
+  template <typename F>
+  auto operator() (F block) const -> decltype(block()) {
+    if (_db) {
+      [_db startTransaction];
+    }
+    auto result = block();
+    if (_db) {
+      [_db commitTransaction];
+    }
+    return result;
+  }
+  __weak id<FSTTransactional> _Nullable _db;
```

Comment:
I'm surprised `_Nullable` is required here. I would have thought `__weak` implied `_Nullable`.

Reply:
Ah, you are correct. I had the `_Nullable` annotation before I realized the need for `__weak`. Fixed.